modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/MongoDataStore.java

```diff
@@ -219,8 +219,7 @@ public void createSchema(SimpleFeatureType incoming) throws IOException {
         incoming.getUserData().put(KEY_collection, incoming.getTypeName());
         
         // Collection needs to exist (with index) so that it's returned with createTypeNames()
-        dataStoreDB.createCollection(incoming.getTypeName(), new BasicDBObject())
-                .ensureIndex(new BasicDBObject(geometryMapping, "2dsphere"));
+        createCollection(incoming.getTypeName(), new BasicDBObject()).ensureIndex(new BasicDBObject(geometryMapping, "2dsphere"));
```

Comment:
Yup, I forgot something when I was fixing the GeoTools version of this branch - {{dataStoreDB.createCollection( .... )}} is correct